### PR TITLE
chore(addons): merge two cfn template's Metadata section

### PR DESCRIPTION
**Pull request by efekarakus**
Merges the ["Metadata"](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/metadata-section-structure.html) section of two cfn templates under "addons/".

Related to #994 

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/addons/cloudformation.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	"gopkg.in/yaml.v3"
+)
+
+type cfnSection int
+
+const (
+	metadataSection cfnSection = iota + 1
+)
+
+// cloudformation represents a parsed YAML AWS CloudFormation template.
```
Comment (iamhopaul123) [inline internal/pkg/addons/cloudformation.go:18]: What do you think of renaming to "cfnTemplate" which will be more informative and avoiding struct and pkg sharing the same name `cloudformation`?

File: internal/pkg/addons/cloudformation.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+)
+
+type cfnSection int
+
+const (
+	metadataSection cfnSection = iota + 1
```
Comment (iamhopaul123) [inline internal/pkg/addons/cloudformation.go:15]: nice! good trick

File: go.mod [hunk 3]
```diff
@@ -29,27 +29,27 @@ require (
...
 	github.com/spf13/viper v1.7.0
 	github.com/stretchr/testify v1.6.0
-	golang.org/x/crypto v0.0.0-20200510223506-06a226fb4e37 // indirect
-	gopkg.in/ini.v1 v1.57.0
-	gopkg.in/yaml.v3 v3.0.0-20200506231410-2ff61e1afc86
 	github.com/syndtr/gocapability v0.0.0-20180916011248-d98352740cb2 // indirect
+	golang.org/x/crypto v0.0.0-20200510223506-06a226fb4e37 // indirect
 	golang.org/x/text v0.3.2 // indirect
+	gopkg.in/ini.v1 v1.57.0
+	gopkg.in/yaml.v2 v2.3.0
```
[the remark this answers is not in the document]
Comment (kohidave) [reply go.mod:51]: Is this because of mergo?

[the remark this answers is not in the document]
Comment (efekarakus) [reply go.mod]: I have no idea 🤷‍♂️ I ran `go mod tidy` again and cleaned up a bunch of old dependencies.
I wonder if I can get dependabot to run tidy periodically 💭

File: internal/pkg/addons/cloudformation.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/addons/cloudformation.go]: Done!

**Comment by efekarakus** [comment]
I updated the PR with an additional function `isEqual` instead of using `reflect.DeepEqual`.

`reflect.DeepEqual` returned false if two yaml nodes had the same content but had different comments, style, or indentation.
`isEqual` compares all the values except those stylistic ones.

**Pull request merged**